src/sources/browser.ts

```diff
@@ -0,0 +1,18 @@
+import { BrowserEngineKind, BrowserKind } from '../types'
+import { getBrowserEngineKind, getBrowserKind, getDocumentFocus, isAndroid } from '../utils/browser'
+
+export default function getBrowserInformation(): BrowserInformation {
+  return {
+    browserEngineKind: getBrowserEngineKind(),
+    browserKind: getBrowserKind(),
+    isAndroid: isAndroid(),
+    documentFocus: getDocumentFocus(),
+  }
+}
+
```

Comment:
IMHO sources should be as atomic as possible to facilitate tree-shaking and simple detectors code. I.e. this source should be split into 4 sources.

@r-valitov @xnerhu What do you think?

Reply:
[detectPluginsLengthInconsistency](https://github.com/fingerprintjs/BotD/blob/2056ef48dbfc4877cb4d2bbed40b2f9d2a1271a3/src/detectors/plugins_inconsistency.ts#L3) would need to check the state of 4 different components then 🤷

Reply:
Worth it, I think. A small helper function can reduce the cumber.

BTW splitting allows sources to fail while affecting a smaller part of the algorithm.